cmake/HalideGeneratorHelpers.cmake

```diff
@@ -57,6 +57,16 @@ function(add_halide_generator TARGET)
         add_executable(${gen} ALIAS ${TARGET})
         target_link_libraries(${TARGET} PRIVATE Halide::Generator ${ARG_LINK_LIBRARIES})
 
+        # For crosscompiling builds, the Halide headers will be included using -isystem,
+        # which will cause all warnings to be ignored. This is not helpful, since
+        # we *want* deprecation warnings to be propagated. So we must set
+        # NO_SYSTEM_FROM_IMPORTED in order for it to be seen.
+        set_target_properties(${TARGET} PROPERTIES NO_SYSTEM_FROM_IMPORTED YES)
+        target_compile_options(${TARGET} PRIVATE
+            $<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:-Wdeprecated-declarations>
+            $<$<CXX_COMPILER_ID:MSVC>:/w14996>  # 4996: compiler encountered deprecated declaration
+        )
+
```

Comment:
I would like these to be opt-out. There should be a bit of code like this:

```
if (NOT ARG_NO_DEFAULT_FLAGS AND NOT Halide_NO_DEFAULT_FLAGS)
    # code from PR
endif ()
```

There are several reasons for this, but the biggest is that `NO_SYSTEM_FROM_IMPORTED` is _very_ coarse, and users who link to another library in addition to Halide (say they want to use Boost or Abseil or fmt or something) for a generator will start getting warnings from those libraries.